examples/plugins/slack/notifications.py

```diff
@@ -0,0 +1,238 @@
+
+    just_date = comparison_date.date()
+    today = date.today()
+
+    prev_monday = today - timedelta(days=today.weekday())
+
+    if owner:  # 'round' dates to previous Monday
+        if prev_monday <= just_date and just_date < prev_monday + timedelta(weeks=1):
+            return "this week"
+        elif prev_monday + timedelta(weeks=1) <= just_date and just_date < prev_monday + timedelta(weeks=2):
+            return "next week"
+        elif prev_monday + timedelta(weeks=2) <= just_date and just_date < prev_monday + timedelta(weeks=3):
+            return "in two weeks"
+        elif prev_monday + timedelta(weeks=3) <= just_date and just_date < prev_monday + timedelta(weeks=4):
+            return "in three weeks"
+        elif prev_monday + timedelta(weeks=4) <= just_date and just_date < prev_monday + timedelta(weeks=5):
+            return "in four weeks"
+    else:
+        if just_date == today + timedelta(days=1):  # one day before
+            return "tomorrow"
+        elif just_date == today + timedelta(weeks=1):  # one week before
+            return "in a week"
+        elif just_date == today + timedelta(weeks=2):  # 2 weeks before
+            return "in two weeks"
+
+    return "soon"
+
+
+@notification_hook_impl
+def access_request_created(
+    access_request: AccessRequest, group: OktaGroup, requester: OktaUser, approvers: list[OktaUser]
+) -> None:
+    """Notify all the approvers of the access request through a notification"""
+    type_of_access = "ownership of" if access_request.request_ownership else "membership to"
+    access_request_url = get_base_url() + f"/requests/{access_request.id}"
+
+    approver_message = (
+        f"{requester.email} has requested {type_of_access} {group.name}.\n\n"
+        f"[View request to approve or reject]({access_request_url})\n\n"
+    )
+
+    # Send the message to both the default channel and the approvers
+    logger.info(f"Approver message: {approver_message}")
+    slack = get_slack_client()
+    slack.send_message(message=approver_message)  # Sends to default channel
+    for approver in approvers:
+        slack.send_message(message=approver_message, user_email=approver.email)
+
+
+@notification_hook_impl
+def access_request_completed(
+    access_request: AccessRequest,
+    group: OktaGroup,
+    requester: OktaUser,
+    approvers: list[OktaUser],
+    notify_requester: bool,
+) -> None:
+    """Notify the requester that their access request has been processed."""
+    access_request_url = get_base_url() + f"/requests/{access_request.id}"
+
+    requester_message = (
+        f"Request for access to {group.name} has been {access_request.status.lower()}.\n\n"
+        f"[View request]({access_request_url})"
+    )
+
+    # Send the message to both the default channel and the requester
+    logger.info(f"Requester message: {requester_message}")
+    slack = get_slack_client()
+    slack.send_message(message=requester_message)  # Sends to default channel
```

Comment:
this will fail if a default channel isn't configure.  i'd really prefer this be configurable.